packages/core-v2-sdk/src/calculateTradePrincipalTokens/calculateTradePrincipalTokens.ts

```diff
@@ -0,0 +1,16 @@
+import {
+  BigNumber,
+  BigNumberish,
+  ContractReceipt,
+  ContractTransaction,
+  Overrides,
+  Signer,
+} from "ethers";
+
+// this is likely to get a lot more complicated with swap kinds, exact in, exact out etc
+export function calculateTradePrincipalTokens(
```

Comment:
doc block here, probably good to explain that the arrays must be same length

Reply:
I'm not sure what u mean by one-stop shop.  yeah only traded against base asset, specifically in our CC Pools, so maybe something like `calcSwapConvergentCurvePool` or something would make the most sense.